File: 27/homework(subsequences)/3/3.py
def taskA():
    with open('27A.txt', 'r', encoding='utf-8') as file:
        info = int(file.readline())
        l = [int(i) for i in file]
        count = 0
        
        for i in range(info):
            k = 0
            for j in range(i, info):
                if l[j] % 5 == 0: k += 1
                if k % 11 == 0: count += 1
        print(count)


def taskB():
    with open('27B.txt', 'r', encoding='utf-8') as file:
        info = int(file.readline())
        l = [0]*11  # отбор по количеству таких чисел, которые кратны 5

        s = 0
        count = 0
        k = 0
        for i in range(info):
            num = int(file.readline())

            if num % 5 == 0: k += 1
            if k % 11 == 0: count += 1
            count += l[k % 11]

            l[k % 11] += 1
        print(count)


if __name__ == '__main__': taskA(); taskB()

File: 27/homework(subsequences)/3/3.cpp
#include <iostream>
#include <vector>
#include <fstream>
#include <ctime>
using namespace std;

int main() {
	int n;
	fstream f("27B.txt");
	f >> n;
	
	vector <int> l(n);
	for(int i = 0; i < n; i++) f >> l[i];
	
	long long count = 0;
	for(int i = 0; i < n; i++) {
		long long k = 0;
		for(int j = i; j < n; j++) {
			if (l[j] % 5 == 0) k++;
			if (k % 11 == 0) count++;
		}
	}
	cout << count << endl;
	cout << clock();
	return 0;
}
